src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -282,8 +282,8 @@ static void checkAttributes(AnonymousFunctionExpressionSyntax syntax, SyntaxList
             MessageID.IDS_FeatureLambdaReturnType.CheckFeatureAvailability(diagnostics, syntax);
 
             Debug.Assert(syntax is not ScopedTypeSyntax);
-            syntax = syntax.SkipScoped(out _).SkipRef(out RefKind refKind);
-            if ((syntax as IdentifierNameSyntax)?.Identifier.ContextualKind() == SyntaxKind.VarKeyword)
+            syntax = syntax.SkipScoped(out _).SkipRef(diagnostics, out RefKind refKind);
```

Comment:
here's an example of where we actually use `RefKind`.  as such, this feature is one that supports such that construct, and so we need to do the lang-ver check to make sure it's legal in this version.